dias/tracker.py

```diff
@@ -403,3 +385,89 @@ def register_done(self, dias_task_name, list_of_files):
             # SQLite limits bulk inserts to 100 at a time
             for batch in chunked(records, 100):
                 Processed.insert_many(batch).on_conflict_replace().execute()
+
+    def add_output_file(self, dias_task_name, start, end, filepath):
+        """Add row to file index table.
+
+        Update the file index table with a row that
+        contains a span of time, the absolute path to the
+        output file and and the span of time the file contains.
+
+        Parameters
+        ----------
+        dias_task_name : String
+            Name of task whose files you wish to track.
+        start : datetime.datetime or Float
+            Earliest time contained in the file.
+        end : datetime.datetime or Float
+            Latest time contained in the file.
+        filepath : str
+            Absolute path to the file
+        """
+        analyzer_id, _ = FileType.get_or_create(name=dias_task_name)
+
+        start = self._ensure_time_unix(start)
+        end = self._ensure_time_unix(end)
+
+        with db:
+            File.insert(
+                {
+                    "file_type_id": analyzer_id,
+                    "filepath": filepath,
+                    "start_time": start,
+                    "end_time": end,
+                    "exists": True,
+                }
+            ).on_conflict_replace().execute()
+
+    def get_output_files(self, dias_task_name, start, end=None, update=True):
+        """Return list of filepaths for dias_task_name that include data between start and end.
+
+        Parameters
+        ----------
+        dias_task_name : String
+            Name of task whose output files you wish.
+        start : datetime.datetime or Float
+            Earliest time contained in the file.
+        end : datetime.datetime or Float
+            Latest time contained in the file.
+        update : bool
+            If true, update the File table before querying
+        """
+        if update:
+            self.update_file_table
+
+        dias_task_analyzer, _ = FileType.get_or_create(name=dias_task_name)
```

Comment:
Should this just be `get` or do we really want to create this here (ine)?

Reply:
Oh, you are totally right. Here, we want to send off an error if there are no files for that analyzer.